### DISCUSSION TITLE
How should Coverlet be configured if I want to display coverage in VSCode?

### DISCUSSION THREAD

**Discussion by bennage** [coverlet-coverage/coverlet#1574] [Q&A] [answered]
My goal is to update coverage info and have it accessible whenever tests are ran in Visual Studio Code. (This is independent of code coverage in our CI/CD process.)

I'm using the [_.NET Test Explorer_](https://github.com/formulahendry/vscode-dotnet-test-explorer) and [_Koverage_](https://github.com/tenninebt/vscode-koverage) extensions, in a devcontainer with .NET 8.
_.NET Test Explorer_ is configured such that tests are executed like: 
`dotnet test --collect:"XPlat Code Coverage;Format=lcov" --no-build --logger "trx;LogFileName=/tmp/some_log.trx"`

Every time the tests are ran, I get a new instance of `coverage.info` inside a folder named with a guid:
`./TestResults/<guid>/coverage.info.`

Based on what I've read, this seems to be the expected behavior however the _Koverage_ extension is confused when there are multiple copies of `coverage.info`.

**Can I configure Coverlet to overwrite `coverage.info` such that there's only a single copy?** 
Or is there a better approach than what I'm doing?

**Comment by Bertk** [comment]
Please use `--results-directory` option (see [dotnet test command](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-test))

> --results-directory <RESULTS_DIR>
> The directory where the test results are going to be placed. If the specified directory doesn't exist, it's created. The default is TestResults in the directory that contains the project file.

e.g.
--results-directory:"$(Build.SourcesDirectory)\artifacts\reports”

**Reply by bennage** [reply]
Thank you!

**Reply by bennage** [reply]
I just tested this, and it's not actually what I'm looking for.
The option allows me produce coverage files at `./artifacts/reports/<guid>/coverage.info` instead of `./TestResults/<guid>/coverage.info`. I was hoping to overwrite previously created coverage files.

Perhaps I am approaching the problem in the wrong way.
Is there any guidance on how to have coverage updated in vscode after tests are executed?

**Comment by Bertk** [comment, marked as answer]
coverlet.msbuild.taks allows to set the coverage file path. https://github.com/coverlet-coverage/coverlet/blob/37f6b8c52b6ae3c122f9794a7adbecc4104bb2ec/Documentation/MSBuildIntegration.md?plain=1#L53-L59

https://github.com/coverlet-coverage/coverlet/blob/37f6b8c52b6ae3c122f9794a7adbecc4104bb2ec/Documentation/MSBuildIntegration.md?plain=1#L41-L45

coverage.collect does not support specifying the coverage file path.

https://github.com/coverlet-coverage/coverlet/blob/37f6b8c52b6ae3c122f9794a7adbecc4104bb2ec/Documentation/VSTestIntegration.md?plain=1#L65-L67

related issue: #1276

**Comment by Bertk** [comment]
This discussion is obsolete.
